### DISCUSSION TITLE
Offer of Help

### DISCUSSION THREAD

**Discussion by odflor** [spotlessmind1975/ugbasic#665] [General]
I am currently learning how to use ugBASIC correctly.

I note that it has very interesting features that facilitate the development of games/demos on various 8-bit platforms.

Sadly, many of the demo programs do not work on my VICE 3.6.1 on the c128, c128z, vic 20, c64 and plus/4 platforms, which are the ones that interest me.

I notice that there are many requests for improvements and bug fixes on the list.

I offer to test all the example programs and indicate which ones have problems and what type.

This is only if I am not going to harm the excellent work that is currently being done in stabilizing this promising tool/programming language.

Greetings and good wishes.

**Comment by spotlessmind1975** [comment]
Hi @odflor, first of all thank you for the interesting idea and for the generous proposal, you are very kind! I'm not sure how you could harm my work. Indeed, as we demonstrated by collaborating on the "What a Wonderful World" demo ( issue #632 ) it seems to me that we have instead improved the compiler in the Windows edition. **So, your help is really welcome!**

Having said that, I think learning to master a language is a wonderful initiative, and I thank you for underlining the advantages of adopting it. Since you report that some example programs don't work on VICE, we first need to understand whether or not these programs should work.

The ugBASIC language is not the same as modern languages, such as C or Pascal. In these languages, the program can rely on "common" behavior among all computers, as if you were programming an abstract computer. In contrast, ugBASIC is an isomorphic language. This means that the same program may or may not run on different computer, depending on the available hardware.

So, first of all, you should check if on the website page corresponding to the example, the Commodore 128 (in the two CPUs), the Commodore VIC-20, the Commodore 64 and the Commodore PLUS/4 are mentioned. If they are not, that example program may not work. So if it doesn't work, it's not really a bug -- or, rather, you (and I, too!) might want it to work, but that's a different story, it's more of an improvement. If it is mentioned and doesn't work, then it's a bug and needs to be fixed. Commodore PLUS/4 is the hardware that gives me the most problems. Anyway, **I consider this type of activity very important**, and if there was a team of people who could take care of it, it would be the best!

Regarding requests for improvements and bug fixes, please note that I am trying to create a sample program for each one (see issue #661 
 and `maths_relative_03.bas` program), so that we can then test and avoid regressions. It's not an easy task, since I'm alone writing this, but if you could help me in testing on your interested targets when they are published, that would be great!

Thank you again!

_ps. for convenience I have moved your report here, as a discussion, so that others can participate too._
